packages/material-ui/src/Badge/Badge.js

```diff
@@ -293,7 +293,10 @@ Badge.propTypes = {
    * The color of the component. It supports those theme colors that make sense for this component.
    * @default 'default'
    */
-  color: PropTypes.oneOf(['default', 'error', 'primary', 'secondary']),
+  color: PropTypes /* @typescript-to-proptypes-ignore */.oneOfType([
+    PropTypes.oneOf(['default', 'error', 'primary', 'secondary']),
+    PropTypes.string,
+  ]),
```

Comment:
Sorry for being offtopic but I just realized why the documentation lists `string` now. We have to widen this to `PropTypes.string` in case new variants are defined in the theme at runtime

I think the better solution is to write a custom validator that compares the passed `color` to the colors available in the theme. And then we need to add another handler for that custom validator in https://github.com/mui-org/material-ui/blob/8e1e856974519472db09ae1d50378eb631eaf2b3/docs/src/modules/utils/generatePropTypeDescription.ts#L69-L97.

Something like `valueFromTheme('color', ['error', 'primary', 'secondary'])` and let `yarn proptypes` insert the default values.

Not required for this to get merged but something to keep in mind.

Reply:
I like this. Will add it on my list for next week. Not sure if I will manage to finish it, but at least to take a look on it. Thanks for sharing 👍

Reply:
How will we access the theme?

Reply:
Yeah I was confused about the same thing. Maybe we can use the options at least in the documentation and validate in the typeChecker just that the value is string..